pennylane/gradients/parameter_shift.py

```diff
@@ -392,33 +439,38 @@ def var_param_shift(tape, argnum, shifts=None, gradient_recipes=None, f0=None):
         gradient_tapes.extend(pdA2_tapes)
 
     def processing_fn(results):
-        # HOTFIX: Apply the same squeezing as in qml.QNode to make the transform output consistent.
-        # pylint: disable=protected-access
-        if tape._qfunc_output is not None and not isinstance(tape._qfunc_output, Sequence):
-            results = [qml.math.squeeze(res) for res in results]
+        if not qml.active_return():
+            # HOTFIX: Apply the same squeezing as in qml.QNode to make the transform output consistent.
+            # pylint: disable=protected-access
+            if tape._qfunc_output is not None and not isinstance(tape._qfunc_output, Sequence):
+                results = [qml.math.squeeze(res) for res in results]
 
         # We need to expand the dimensions of the variance mask,
         # and convert it to be the same type as the results.
         res = results[0]
-        ragged = getattr(results[0], "dtype", None) is np.dtype("object")
+
 
         mask = []
         for m, r in zip(var_mask, qml.math.atleast_1d(results[0])):

```

Comment:
So here we only post process the `var` part of the results? But all results are computed by `var_param_shift` right?

Reply:
Yes. `var_param_shift` manages which part of the result can be computed as expvals (involutory obs) and which require the extra computations.